src/js/components/InfiniteScroll/InfiniteScroll.js

```diff
@@ -8,158 +8,171 @@ class InfiniteScroll extends PureComponent {
   };
 
   static getDerivedStateFromProps(nextProps, prevState) {
-    const { show, step } = nextProps;
+    const { items, show, step } = nextProps;
+    const pageCount = Math.ceil(items.length / step);
     if (
       prevState.firstPage === undefined ||
-      (show && show >= step * (prevState.lastPage + 1))
+      ((show && show >= step * (prevState.lastPage + 1)) ||
+        pageCount !== prevState.pageCount)
     ) {
       let lastPage = prevState.lastPage || 0;
       if (show && show >= step * (lastPage + 1)) {
         lastPage = Math.floor((show + step) / step) - 1;
       }
-      return { firstPage: 0, lastPage };
+      return { firstPage: 0, lastPage, pageCount };
     }
     return null;
   }
 
   state = {};
 
-  beginPageRef = React.createRef();
-
-  endPageRef = React.createRef();
-
-  showRef = React.createRef();
```

Comment:
I'd love to discuss these changes on our call today.

Reply:
done